doc/devices/dts/device_tree.rst

```diff
@@ -80,6 +80,59 @@ This .fixup file by default resides in the board directory and is named
 dts.fixup.  This fixup file maps the generated include information to the
 current driver/source usage.
 
+.. _dt_vs_kconfig:
+
+Device Tree vs Kconfig
+**********************
+
+As mentioned above there are several tools used to configure the build with
+Zephyr.
+The two main ones, Device Tree and Kconfig, might seem to overlap in purpose,
+but in fact they do not. This section serves as a reference to help you decide
+whether to place configuration items in Device Tree or Kconfig.
+
+The scope of each configuration tool can be summarized as follows:
+
+* Device Tree is used to describe the **hardware** and its **boot-time
+  configuration**.
+* Kconfig is used to describe which **software features** will be built into
+  the final image, and their **configuration**.
+
+Hence Device Tree deals mostly with hardware and Kconfig with software. A
+noteworthy exception is Device Tree's ``chosen`` keyword, which allows the user
+to select a particular instance of a hardware device to be used for a concrete
+purpose by the software. An example of this is selecting a particular UART for use
+as the system's console.
+
+To further clarify this separation, let's use a particular, well-known
+example to illustrate this: serial ports a.k.a. UARTs. Let's consider a
+board containing a SoC with 2 UART instances:
+
+* The fact that the target hardware **contains** 2 UARTs is described with Device
+  Tree. This includes the UART type, its driver compatibility and certain
+  immutable (i.e. not software-configurable) settings such as the base address
+  of the hardware peripheral in memory or its interrupt line.
+* Additionally, **hardware boot-time configuration** is also described with Device
+  Tree. This includes things such as the interrupt line and IRQ priority. These
```

Comment:
That's a very particular example, changing irq/prio at runtime is going to be very rare. Use the i2c clock-frequency instead, since it's very common to change it at runtime.

Reply:
I've amended this now using UART baudrate